### PR TITLE
Add recipe for Qt Location

### PR DESCRIPTION
**Pull request by ceholden**
Adds recipe for Qt Location, a Qt library that assists with spatial data (map providers, spatial data, GPS, etc). This recipe is being added primarily as a dependency for upgrading QGIS to version 3 (see https://github.com/conda-forge/qgis-feedstock/issues/17)

### CODE REVIEW

File: recipes/qtlocation/bld.bat [hunk 1]
```diff
@@ -0,0 +1,28 @@
+@echo on
+mkdir build
+if errorlevel 1 exit /B 1
+cd build
+if errorlevel 1 exit /B 1
+
+qmake ..\qtlocation.pro
+if errorlevel 1 exit /B 1
+
+nmake
```
Comment (xoviat) [inline recipes/qtlocation/bld.bat]: Please use `jom -j%CPU_COUNT%` and add jom to requirements

Comment (ceholden) [reply recipes/qtlocation/bld.bat]: Sounds good! Thank you for the review

**Comment by ceholden** [comment]
Any clues why conda build is trying to build the recipe three times? I'll need to silence the entire build process for OS X to keep the logfile under 4MB if it tries to build more than once. Seems to be happening in other PRs as well. Good news is it looks like all platforms built correctly

```
Resolving distributions from 1 recipes... 
Computed that there are 3 distributions from the 1 recipes:
Resolved dependencies, will be built in the following order: 
	qtlocation-5.6.2-0 (will be built: True)
	qtlocation-5.6.2-0 (will be built: True)
	qtlocation-5.6.2-0 (will be built: True)
```

**Comment by ceholden** [comment]
Assuming the 3 distributions from 1 recipe is normal, as explained in the ["Common Issues" Wiki entry](https://github.com/conda-forge/staged-recipes/wiki/Common-issues#why-is-staged-recipes-building-my-package-3-times-when-python-is-only-a-build-dependency), and that it won't be a worry once it becomes a feedstock.

I've silenced all the stdout from `make`, `make check`, and `make install` for OSX builds for now to avoid
```
The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).
The job has been terminated
```
🤞

**Comment by ceholden** [comment]
Rebased against master, squashed my commits down to 1, and fixed up a few remaining things I noticed (not pinning perl, preventing duplicate vc14 builds on Windows with py35/py36).

Would someone mind giving it a lookover and merging?

Thanks!

**Comment by gillins** [comment]
@ocefpaf another one for you to merge if you are happy to. We need this for building `qgis` ...

**Comment by gillins** [comment]
Thanks!

**Comment by ocefpaf** [comment]
No problem. Sorry it took me so long to see this one. Please feel free to ping me on PRs like these.

**Pull request merged**